### PR TITLE
Parser/67 rebase parser

### PR DESCRIPTION
**Pull request by osvegn**
Pull request link to the issue: #67.

### CODE REVIEW

File: server/src/parser/call_command.c [hunk 1]
```diff
@@ -5,27 +5,19 @@
...
 */
 
-/// \file server/src/parser/call_commands.c
-
 #include "cli_commands.h"
-#include <stddef.h>
-#include <stdbool.h>
 
-int call_command(char **arguments)
+int call_command(cli_command_t *command)
```
Comment (LucasTesnier) [inline server/src/parser/call_command.c:10]: Where is the call ?

File: server/src/parser/fill_command_arguments.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    for (int i = 0; i < arg_number; i++) {
+        if (command[0] != ' ' || command[1] != '"')
+            return (NULL);
+        command = command + 2;
+        for (arg_len = 0; command[arg_len] && command[arg_len] != '"';
+        arg_len++);
+        if (!command[arg_len])
+            return (NULL);
+        args[i] = malloc(sizeof(char) * (arg_len + 1));
```
Comment (LucasTesnier) [inline server/src/parser/fill_command_arguments.c:36]: Malloc not verified

File: server/src/parser/fill_command_arguments.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        strncpy(args[i], command, arg_len);
+        command = command + arg_len + 1;
+    }
+    if (command[0])
+        return (NULL);
+    args[arg_number] = NULL;
+    return (args);
+}
+
+static char **get_arguments(char *command, int arg_number)
```
Comment (LucasTesnier) [inline server/src/parser/fill_command_arguments.c:46]: Missing documentation

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
NTR!

File: tests/unit_tests/parser/compute_command_test.c [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+*/
+
+#include <criterion/criterion.h>
+#include "parser.h"
+#include "cli_commands.h"
+
+Test(compute_command, test01)
+{
+    cr_assert_eq(1, 1);
+}
```
Comment (Cavonstavant) [inline tests/unit_tests/parser/compute_command_test.c:15]: tbr

**Review by Cavonstavant — CHANGES_REQUESTED** [review]

**Review by LucasTesnier — APPROVED** [review]

**Review by Cavonstavant — APPROVED** [review]

**Pull request merged**